### ISSUE TITLE
TdS Validator v1.1.5 panic in Uncompressed::to_slots

### ISSUE THREAD

**Issue by svenski123** [solana-labs/solana#9686]
Solana validator v1.1.5 running in TdS was panicked by BitVec::get_bit in solana_core::epoch_slots::Uncompressed::to_slots.

The panic halted normal processing but the process did not exit and began leaking memory until killed.

```
[2020-04-23T09:47:45.567083762Z INFO  solana_core::cluster_info] ReceiveUpdates took: 188 ms  len: 1
thread 'solana-repair-service' panicked at 'BitVec::get_bit: out of bounds', <::std::macros::panic macros>:2:4
stack backtrace:
[2020-04-23T09:47:45.771207067Z INFO  solana_core::cluster_info] ReceiveUpdates took: 101 ms  len: 1
[... elided ...]
[2020-04-23T09:47:47.998801978Z INFO  solana_core::cluster_info] ReceiveUpdates took: 103 ms  len: 2
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:77
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1052
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1426
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:204
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:224
  10: solana_metrics::metrics::set_panic_hook::{{closure}}::{{closure}}
  11: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:476
  12: std::panicking::begin_panic
  13: solana_core::epoch_slots::Uncompressed::to_slots
  14: solana_core::epoch_slots::CompressedSlots::to_slots
  15: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter
  16: solana_core::cluster_slots::ClusterSlots::update
  17: solana_core::repair_service::RepairService::run
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

No further log output was produced however the validator process continued running for another 19 minutes until it was subsequently killed by kernel OOM killer.
```
# dmesg -T
...
[Thu Apr 23 10:06:11 2020] Out of memory: Kill process 24830 (solana-validato) score 715 or sacrifice child
[Thu Apr 23 10:06:11 2020] Killed process 24830 (solana-validato) total-vm:200723364kB, anon-rss:49682036kB, file-rss:0kB, shmem-rss:0kB
...
```

**Comment by leoluk** [comment]
This is a duplicate of https://github.com/solana-labs/solana/issues/9591

Looks like I accidentally triggered it while exploiting https://github.com/solana-labs/solana/issues/9684?

**Comment by svenski123** [comment]
The BitVec::get_bit panic piece looks the same - I'll take your word that you caused it. :)

The fact that the program didn't exit on panic but continued running zombie-like and leaking memory is concerning - this is twice now a panicked validator nearly brought the box down.

I do have a 47GB core file and objects so perhaps the are clues there as to what hung it up - obviously getting it not to panic in the first place is the priority though.

I thought it better to post logs, stack dumps, etc. here rather than clog up discord.

**Comment by sakridge** [comment]
Yea, we should probably call `process::exit` in our panic handler. @mvines what do you think?

**Comment by mvines** [comment]
> Yea, we should probably call `process::exit` in our panic handler. @mvines what do you think?

yeah totally, I'll take that

**Comment by mvines** [comment]
https://github.com/solana-labs/solana/issues/9591 was fixed in v1.1.6 so the fact that it reproduces on v1.1.5 is a known issue.  Fix: update to v1.1.6.  Looks like there's nothing more to do here so closing the issue
